services/src/main/java/org/exoplatform/task/rest/TaskRestService.java

```diff
@@ -481,6 +482,21 @@ public Response getLabels() {
         return Response.serverError().entity(e.getMessage()).build();
         }
   }
+  @GET
+  @Path("labels/project/{id}")
+  @RolesAllowed("users")
+  @Produces(MediaType.APPLICATION_JSON)
+  @ApiOperation(value = "Gets labels of the authenticated user", httpMethod = "GET", response = Response.class, notes = "This returns labels of the authenticated user")
+  @ApiResponses(value = { @ApiResponse(code = 200, message = "Request fulfilled") })
+  public Response getLabelsByProjectId(@ApiParam(value = "project id", required = true) @PathParam("id") long id) {
+    try {
```

Comment:
I think a check is missing to ensure the current user has access to the project (a user not member of a project should not be able to list its labels)